src/services/vector_store/pinecone_store.py

```
@@ -0,0 +1,139 @@
+                    name=self.index_name,
+                    dimension=300,  # spaCy medium model dimension
+                    metric="cosine",
+                )
+                print(f"✅ Created Pinecone index: {self.index_name}")
+
+            # Connect to index
+            self.index = pinecone.Index(self.index_name)
+            print(f"✅ Connected to Pinecone index: {self.index_name}")
+
+        except Exception as e:
+            raise ConfigurationError(f"Failed to initialize Pinecone: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)